compiler/cpp/src/util/JSONParser.cpp

```diff
@@ -1,10 +1,14 @@
-//
-// Created by Marc on 24.05.2021.
-//
+// Copyright (c) 2021 Compose Generator Contributors. All rights reserved.
 
 #include "JSONParser.h"
 
+#include <utility>
+
 json JSONParser::getJSONValueFromKey(const std::unique_ptr<KeyExprAST>& key) {
+    return getJSONValueFromKey(data, key);
+}
+
+json JSONParser::getJSONValueFromKey(const json data, const std::unique_ptr<KeyExprAST>& key) {
```

Comment:
Given:

```javascript
"obj": {
   "a": 42,
   "b": {
       "c:" 43
    }
}
```

This aproach doesn't allow something like: `obj contains c == 43`, right? "Just" `obj contains b.c == 43` is possible, right? But I don't think that one needs to allow for recursive descent or do you have that use case in Compose Generator? It would also make things more complicated since variables can now occur multiple times, e.g. in

```javascript
"obj": {
   "a": 42,
   "b": {
       "c:" 43
    },
   "d": {
       "c": 143
    }
}
```

Reply:
No, the current approach does not support `obj contains c == 43` in your example and recursive descent would potentially cost a lot of performance.
And yes, there is currently no usecase for Compose Generator, so I think its okay to leave it like it is for the moment. If, and as soon as we need it, we could think about adding a new statement like `search` for that, so we can choose to search on the first level of the tree or we want to search in the whole sub-tree ...